include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -476,26 +483,110 @@ class alignment_file_input_format_REMOVEME<format_sam>
+        {
+            case 'M': case '=': case 'X': ref_length += cigar_count, seq_length += cigar_count; break;
+            case 'D': case 'N':           ref_length += cigar_count; break;
+            case 'I' :                    seq_length += cigar_count; break;
+            case 'S': case 'H': case 'P': break; // no op (soft-clipping or padding does not increase either length)
+            default: throw format_error{"Illegal cigar operation: " + std::string{cigar_operation}};
+        }
+    };
+
+    /*!\brief Parses a cigar string into a vector of operation-count pairs (e.g. (M, 3)).
+     * \ingroup alignment_file
+     * \tparam cigar_input_type The type of a single pass input view over the cigar string; must model
+     *                          std::ranges::input_range.
+     * \param[in]  cigar_input  The single pass input view over the cigar string to parse.
+     *
+     * \returns A tuple of size three containing (1) std::vector over seqan3::cigar, that describes
+     *          the alignment, (2) the aligned reference length, (3) the aligned query sequence length.
+     *
+     * \details
+     *
+     * For example, the view over the cigar string "1H4M1D2M2S" will return `{[(H,1), (M,4), (D,1), (M,2), (S,2)], 7, 6}`.
+     */
+    template <typename cigar_input_type>
+    [[nodiscard]] std::tuple<std::vector<cigar>, int32_t, int32_t> parse_cigar(cigar_input_type && cigar_input) const
+    {
+        std::vector<cigar> operations{};
+        std::array<char, 20> buffer{}; // buffer to parse numbers with from_chars. Biggest number should fit in uint64_t
+        char cigar_operation{'\0'};
+        uint32_t cigar_count{};
+        int32_t ref_length{}, seq_length{}; // length of aligned part for ref and query
+
+        // transform input into a single input view if it isn't already
+        auto cigar_view = cigar_input | views::single_pass_input;
+
+        // parse the rest of the cigar
+        // -----------------------------------------------------------------------------------------------------------------
+        while (std::ranges::begin(cigar_view) != std::ranges::end(cigar_view)) // until stream is not empty
+        {
+            auto buffer_end = (std::ranges::copy(cigar_view | views::take_until_or_throw(!is_digit), buffer.data())).out;
+            cigar_operation = *std::ranges::begin(cigar_view);
+            std::ranges::next(std::ranges::begin(cigar_view));
+
+            if (std::from_chars(buffer.begin(), buffer_end, cigar_count).ec != std::errc{})
+                throw format_error{"Corrupted cigar string encountered"};
+
+            update_alignment_lengths(ref_length, seq_length, cigar_operation, cigar_count);
+            operations.emplace_back(cigar_count, cigar_op{}.assign_char(cigar_operation));
+        }
+
+        return {operations, ref_length, seq_length};
+    }
+
+    /*!\brief Transfer soft clipping information from the \p cigar_vector to \p sc_begin and \p sc_end.
```

Comment:
```suggestion
    /*!\brief Transfer soft clipping information from the `cigar_vector` to `sc_begin` and `sc_end`.
```

Reply:
I think the \p command does this already?

Reply:
as discussed.